proj/src/main/webapp/script.js

```diff
@@ -27,12 +29,14 @@ if(btnLogin){
     });
 }
 
+var keyIdDict = {};
 
 // Adds user to an existing chat when given a reference to the place in the database
-function addUserToTag(reference){
+function addUserToTag(reference, tag){
     var currentUID = firebase.auth().currentUser.uid;
     console.log("adding new user to chat room with uid: " + currentUID);
-    reference.push(currentUID);
+    var removalKey = reference.push(currentUID).key;
+    keyIdDict[tag] = removalKey;
```

Comment:
Rather than storing it to a dict here and then somewhere else adding the key to the user, can you add it to the user here so that all the functionality of adding a user to a chat is in one method? Although it might require multiple calls to the database rather than one call at the end of everything.

Reply:
Scratch that, I guess that is what `createOrJoinChat` does. But I think this would be better if it returned the key instead of saving it to a global object. Maybe similar to how it is done for `allTags`.